src/ome_zarr_models/v04/multiscales.py

```diff
@@ -1,17 +1,144 @@
-from typing import Any
+from __future__ import annotations
+from collections import Counter
+from typing import Annotated, Any, Sequence, get_args
 
-from pydantic import Field, field_validator
+from pydantic import AfterValidator, Field, model_validator
 
 from ome_zarr_models.base import Base
-from ome_zarr_models.utils import _unique_items_validator
-from ome_zarr_models.v04.axes import Axis
+from ome_zarr_models.utils import duplicates
+from ome_zarr_models.v04.axes import Axis, AxisType
 from ome_zarr_models.v04.coordinate_transformations import (
     PathScale,
     PathTranslation,
     VectorScale,
     VectorTranslation,
 )
 from ome_zarr_models.v04.omero import Omero
+from pydantic_zarr.v2 import ArraySpec, GroupSpec
+import zarr
+
+VALID_NDIM = (2, 3, 4, 5)
+NUM_TX_MAX = 2
+
+
+def ensure_transform_dimensionality(
```

Comment:
Having read this function, to simplify it I would actually decouple the validation from the spec a bit, and write the validation function to check that an arbitrary list of transforms (that therefore have a `.ndim` attribute) have the same number of dimensions.

So:
1. Remove all transforms from the list that are Path transforms
2. If `>= 2` transforms left, check they have same dimensions

I think this will make typing/logic a bit nicer, and also make this a bit more re-usable for whatever comes in coordinate transform changes.

Reply:
good idea! I will implement this